### PR TITLE
[feature] Support submodes for IDFM

### PR DESCRIPTION
**Pull request by woshilapin**

### CODE REVIEW

File: src/documentation/netex_idfm_to_ntfs.md [hunk 1]
```diff
@@ -165,21 +165,26 @@ Be careful, `physical_mode_id` and `commercial_mode_id` are **not** prefixed.
...
+--- | --- | --- | --- | --- | --- | ---
+air | _any_ | Air | Avion | Air | Avion |
+bus | _any_ | Bus | Bus | Bus | Bus |
+coach | _any_ | Coach | Autocar | Coach | Autocar |
+ferry | _any_ | Ferry | Ferry | Ferry | Ferry |
+metro | _any_ | Metro | Métro | Metro | Métro |
+rail | railShuttle | RailShuttle | Orlyval, CDG VAL | RailShuttle | Orlyval, CDG VAL |
+rail | suburbanRailway | LocalTrain | Train Transilien | LocalTrain | Train Transilien |
+rail | regionalRail | Train | TER Intercités | regionalTrain | TER |
+rail | interregionalRail | Train | TER Intercités | interregionalTrain | Intercités |
```
Comment (ArnaudOggy) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
rail | regionalRail | Train | TER / Intercités | regionalTrain | TER |
rail | interregionalRail | Train | TER / Intercités | interregionalTrain | Intercités |
```

Comment (woshilapin) [reply src/documentation/netex_idfm_to_ntfs.md]: Done in f89552b55e25c336cbbbceebb98d74fc12388dd6.

File: src/netex_idf/modes.rs [hunk 2]
```diff
@@ -12,109 +12,282 @@
...
+            IDFMMode::RailRegional,
+            NTFSMode::new(("Train", "TER / Intercités"), ("regionalRail", "TER")),
         );
         m.insert(
-            "trolleyBus",
-            ModeNetexIDF {
-                physical_mode: ("Tramway", "Tramway"),
-                commercial_mode: ("TrolleyBus", "TrolleyBus"),
-            },
+            IDFMMode::RailShuttle,
```
Comment (ArnaudOggy) [inline src/netex_idf/modes.rs]:
```suggestion
            IDFMMode::RailInterregional,
```

Comment (woshilapin) [reply src/netex_idf/modes.rs]: Thanks. Done in 262c93cc6a7f0d2c514fd2ed1ac466210b57b96c.

**Comment by ArnaudOggy** [comment]
Nice :+1:

**Pull request merged**